Sources/StreamChatUI/Utils/LoaderView.swift

```diff
@@ -0,0 +1,307 @@
+            drawPath(context: context, view: arrangedSubview, fixedFrame: frame)
+        }
+    }
+    
+    private func drawCutoutInContainer(
+        view: Container,
+        fromParentView parentView: UIView,
+        context: CGContext?
+    ) {
+        view.arrangedSubviews.forEach { arrangedSubview in
+            if arrangedSubview is Container {
+                recursivelyDrawCutOutInView(arrangedSubview, fromParentView: parentView, context: context)
+                return
+            }
+            let frame = view.convert(arrangedSubview.frame, to: parentView)
+            drawPath(context: context, view: arrangedSubview, fixedFrame: frame)
+        }
+    }
+    
+    private func drawPath(
+        context: CGContext?,
+        view: UIView,
+        fixedFrame: CGRect? = nil
+    ) {
+        let frame = fixedFrame ?? view.frame
+        context?.setBlendMode(.clear)
+        let rect = frame
+        let clipPath: CGPath = UIBezierPath(
+            roundedRect: rect,
+            cornerRadius: frame.height * heightToCornerRadiusRatio
+        ).cgPath
+        context?.addPath(clipPath)
+        context?.setFillColor(UIColor.clear.cgColor)
+        context?.closePath()
+        context?.fillPath()
+    }
+
+    override func layoutSubviews() {
+        setNeedsDisplay()
+        superview?.ld_getGradient()?.frame = (superview?.bounds)!
+    }
+}
+
+extension UIView {
+    private static var cutoutHandle: UInt8 = 0
+    private static var gradientHandle: UInt8 = 0
+    private static var loaderDuration = 0.85
+    private static var gradientWidth = 0.17
+    private static var gradientFirstStop = 0.1
+    
+    private func ld_getCutoutView() -> UIView? {
+        objc_getAssociatedObject(self, &Self.cutoutHandle) as! UIView?
+    }
+    
+    private func ld_setCutoutView(_ aView: UIView) {
+        objc_setAssociatedObject(self, &Self.cutoutHandle, aView, .OBJC_ASSOCIATION_RETAIN)
+    }
+    
+    fileprivate func ld_getGradient() -> CAGradientLayer? {
+        objc_getAssociatedObject(self, &Self.gradientHandle) as! CAGradientLayer?
+    }
+    
+    private func ld_setGradient(_ aLayer: CAGradientLayer) {
+        objc_setAssociatedObject(self, &Self.gradientHandle, aLayer, .OBJC_ASSOCIATION_RETAIN)
+    }
+    
+    fileprivate func ld_addLoader(heightToCornerRadiusRatio: CGFloat) {
+        let gradient: CAGradientLayer = CAGradientLayer()
+        gradient.frame = CGRect(x: 0, y: 0, width: bounds.size.width, height: bounds.size.height)
+        layer.insertSublayer(gradient, at: 0)
+        
+        configureAndAddAnimationToGradient(gradient)
+        addCutoutView(heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+    }
+    
+    fileprivate func ld_removeLoader() {
+        ld_getCutoutView()?.removeFromSuperview()
+        ld_getGradient()?.removeAllAnimations()
+        ld_getGradient()?.removeFromSuperlayer()
+        
+        for view in subviews {
```

Comment:
somewhere is using loop `for view in views`, somewhere is `forEach`, please use consistent approach in all places to make code more readable